HIPs/incubation/web3j.md

```diff
@@ -0,0 +1,206 @@
+- contracts: smart contract specific types
+
+## Smart Contract Testing
+
+The Web3j core library is complemented by the following projects that facilitate unit and integration testing of Solidity smart contracts.
+
+The [Web3j EVM](https://github.com/web3j/web3j-evm) project provides an embedded EVM that allows developers to write applications that simulate transactions taking place on the EVM. 
+
+The EVM contained within the project is the Hyperledger Besu EVM. This ensures that the EVM can be run within the local JVM without the need to orchestrate a full Ethereum node via Docker or similar tools.
+
+One of the significant advantages of using the Hyperledger Besu EVM is that developers work with a production-grade EVM during the development process. In other non-JVM development environments, it's unlikely that developers will have at their disposal an embedded EVM that uses the same code as a widely used Ethereum client.
+
+The Web3j EVM project is complemented by the [Web3j Unit](https://github.com/web3j/web3j-unit) project. This project is used to inject different EVM environments into unit and integration tests.
+
+This allows developers to transact with real Ethereum execution environments from their unit tests. The most common use cases are smart contract deployments and interacting with them, but any type of EVM transaction can be performed.
+
+By default the Web3j unit project uses the embedded Hyperledger Besu EVM for this purpose to ensure fast execution time. However, it is also possible to use containerised instances of Ethereum execution clients such as Hyperledger Besu and Geth for this purpose. This is especially useful for integration testing purposes to simulate production environments.
+
+## Build Plugins
+
+The JVM platform utilised two major build systems — Gradle and Maven. The [Web3j Gradle](https://github.com/web3j/web3j-gradle-plugin) and [Web3j Maven](https://github.com/web3j/web3j-maven-plugin) projects implement support for these build systems.
+
+One of the complexities that developers face when working with smart contract development is the need for a separate Solidity compilation environment. 
+
+When working with these plugins they install the the most recent version of the Solidity compiler that is appropriate for your project, enabling the compilation of any Solidity source files at build time.
+
+Following the Solidity build process, these build plugins generate Java wrappers for integrating with these smart contracts using Web3j.
+
+This enables the development process to remain within the developers JVM environment, resulting in no additional installations apart from the Java Development Kit (JDK) or Android equivalent.
+
+In addition, there may be opportunities to integrate the build plugins with Hyperledger Solang which could be explored.
+
+## CLI
+
+Web3j also comes with a CLI providing a number of different functions including:
```

Comment:
if this will be based on the repo web3j/web3j-cli to be contributed along with the others mentioned above, can you add the link here as was done for the other components?